vignettes/phyloVision.Rmd

```diff
@@ -0,0 +1,151 @@
+
+```
+pip install git+https://github.com/yoseflab/Hotspot.git
+```
+
+# Using PhyloVision
+
+Running an analysis with vision consists of three steps:
+
+1. Creating the PhyloVision object
+2. Running the `phyloAnalyze` function
+3. Running Hotspot
+4. Browsing results
+
+First, we need to load VISION, reticulate and ape.
+```{r setup, eval=F}
+knitr::opts_chunk$set(echo = TRUE)
+library(VISION)
+library(reticulate)
+library(ape)
+```
+
+## Creating a PhyloVision Object with a Tree
+
+Vision objects now support dendrograms for visualization and analysis. To create the PhyloVision object, you need all of the same data, an expression matrix, and metadata and/or signatures, but you also include a tree object from the Ape package.
+
+```{r create, eval=F}
+# Read the expression and meta data
+file_path <- "data/embryogenesis" # Replace the path
+
+expr <- read.table(paste(file_path, "expr_filtered.tsv.gz", sep="/"), check.names = FALSE, sep = "\t")
+
+meta <- read.table(paste(file_path, "meta.tsv", sep="/"), check.names = FALSE, sep = "\t", row.names = 1, header=TRUE)
+meta$Cell_Type <- as.factor(meta$Cell_Type)
+# Signature file
+sig <- paste("data", "h.all.v5.2.symbols.gmt", sep="/")
+
+# Read the tree
+tree <- read.tree(paste(file_path, "embryo_tree.newick", sep="/"))
+# Collapse one mutations
+tree <- collapse.singles(tree)
+expr <- expr[, tree$tip.label]
+
+# Construct the Vision object
+vis <- PhyloVision(tree=tree, data=expr, signatures=sig, meta=meta, num_neighbors=30, projection_genes= rownames(expr))
+```
+**Tree**
+
+The `ape` phylogeny object. Singleton edges should be collapsed prior to use. Ensure that leaf names correspond to same names used in the expression matrix. 
+
+**Expression Data**
+
+The provided expression data should be scaled and normalized. It is recommended to apply more advanced normalization procedures such as batch correction or removal of technical confounders.
+
+The expression data should not be log-transformed prior to loading into VISION.
+
+**Signatures**
+
+Signatures can be provided as a list of paths to signature files (\*.gmt) or Signature objects.
+
+See [the signature vignette](Signatures.html) for more information on finding or creating gene Signatures.
+
+**Meta Data**
+
+An R data.frame with cell-level meta-data.  This could be confounding variables (e.g. percentage of mitochondrial RNA, number of genes detected) or experimental covariates (e.g. genotype, donor, batch).
+
+This input is optional if Signatures are provided.
+
+**Other Options**
+
+Other options and inputs can be provided to customize how VISION runs.
+
+## Running PhyloVision analysis
+
+Next, we can perform the normal Vision analysis using the tree as the latent space.
+
+``` {r analyze, eval=F}
+vis <- phyloAnalyze(vis)
+```
+## Running Hotspot analysis
+
+We can also perform Hotspot module analysis. The expression data is already logged for us.
```

Comment:
The expression data is already logged for us, so.... ?

Can you please flesh this out a bit more? Let's remember that this is for users who may have never used either of these tools and we want to be as clear as possible.

Reply:
I have the link to full hotspot documentation in the running hotspot analysis line (later on)

How does this sound in the intro?

"Hotspot helps find new informative gene sets, 'modules', that are autocorrelated with a specific modality, in this case, the phylogeny. These new gene sets can be compared with the existing signature sets to investigate biological function. "

Reply:
That's better! Let's put the link at the _top_ in the Preliminaries, and when you run Hotspot.

I'd like you take a look at David's documentation here: https://github.com/YosefLab/Hotspot

Please take as much language as possible from this Github repo and his associated notebooks. It's important that we're comprehensive and clear here.